ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,220 @@
+- Users who prefer GitOps principles cannot benefit from UI improvements, creating a permanent split in the user experience
+
+**Chosen approach** replaces both current flows with a unified Git-centric solution that preserves GitOps benefits while providing the usability improvements users expect.
+
+### Core Shift
+- **Replace UI configuration** with a VS Code plugin for onboarding and editing
+- **Use local schema validation**, linting, and previews to reduce onboarding errors
+- **Use Git as the single source of truth** for applications, components and secrets
+- **Preserve the UI for read-only activities** for monitoring, logs, build inspection, and metrics
+- **Commit to single onboarding flow**: UI-based onboarding will be completely deprecated with no fallback option to minimize maintenance overhead and deliver optimal user outcomes
+
+### IDE Selection Rationale
+VS Code was chosen as the primary IDE target because:
+- **Wide adoption**: VS Code is extensively used across development teams
+- **Robust plugin ecosystem**: Provides comprehensive APIs for linting, schema validation, and custom tooling
+- **Advanced visualization capabilities**: Plugin system supports rich visualizations and can run full, sandboxed React-based applications within webview windows
+- **Extensibility**: Enables building sophisticated onboarding wizards and configuration interfaces directly within the familiar IDE environment
+
+### Component Architecture
+
+| Component | Role |
+|-----------|------|
+| VS Code Plugin | Main interface for onboarding and configuration |
+| Git Repos | Source of truth for Konflux config |
+| Vault | Preferred secrets backend |
+| Kubernetes Secrets | Alternative backend for secrets |
+| External Secrets Operator (ESO) | Delivery of secrets to Kubernetes |
+| Konflux UI | Monitoring and runtime insight only |
+| Fleet Registry | Service discovery and instance metadata for multi-instance deployments |
+
+### VS Code Plugin Design
+
+**Core Features:**
+- Forms and wizards to scaffold and edit YAML objects
+- Built-in schema-aware YAML editor
+- Linting and validation against Konflux schemas
+- Integration with local secret store session (Vault or Kubernetes)
+- Git-aware diff view and PR-ready commit generation
+- Optional: preview pipeline graph / dependency resolution
+
+**Secrets UX:**
+- Form-based secret creation (with validation, TTLs, metadata)
+- Generated secret definitions point to secret store paths via ESO
+- Secrets created in the secret store configured for that instance of Konflux
+- Initially supported stores: Vault and Kubernetes Secrets
+- Optional: audit log stored in `.konflux/` for local tracking
+
+**GitOps Flow:**
+1. Developer opens repo in VS Code
+2. Launches Konflux plugin
+3. Uses wizard to configure new component/build
+4. Adds secrets as needed via plugin form
+5. Plugin generates YAML and commits to Git
+6. Developer pushes PR
+7. CI runs lightweight schema validation (should pass if plugin was used)
```

Comment:
Ah, good - CI isn't going away.

Reply:
we can give them the tools for CI but we can't enforce it as in my design, the user owns their own repo.